app/Utilities/twitter_wrapper.rb

```diff
@@ -0,0 +1,47 @@
+class TwitterWrapper
+  include Singleton
+
+  def initialize
+    @client = Twitter::REST::Client.new do |config|
+      config.consumer_key = ENV['TWITTER_CONSUMER_KEY']
+      config.consumer_secret = ENV['TWITTER_CONSUMER_SECRET']
+      config.access_token = ENV['TWITTER_ACCESS_TOKEN']
+      config.access_token_secret = ENV['TWITTER_ACCESS_TOKEN_SECRET']
+    end
+  end
+
+  def get_current_profile_image(uid)
+    if user_valid?(uid)
+      user = fetch_user_by_uid(uid)
+      latest_image = user.profile_image_url
+    else
+      return nil
+    end
+    latest_image
+  end
+
+  private
+
+    def user_valid?(uid)
+      begin
+        check_user_validity(uid)
+      rescue
```

Comment:
this will mark user invalid even if failure is network related. Please add exact exceptions where this should be handled and user marked as invalid.